### ISSUE TITLE
Compilation failures: option CONFIG_AIC7XXX_BUILD_FIRMWARE 

### ISSUE THREAD

**Issue by jualvespereira** [TuxML/ProjetIrma#164]
Investigating the failures due to option `CONFIG_AIC7XXX_BUILD_FIRMWARE`, I came across 4 main errors:

1. `'aicasm_symbol.c:48:19: fatal error: aicdb.h: No such file or directory'`: It's due options `CONFIG_AIC7XXX_BUILD_FIRMWARE == 'y' or CONFIG_AIC79XX_BUILD_FIRMWARE == 'y'`. According to the qualitative analysis of the error message, it requires additional software. Therefore, it seems to be a TUXML bug. (2,395 errors)

1. `'/bin/sh: 1: as68k: not found'`: this option allows you to rebuild firmware run by the QUICC processor. It requires a supplementary tool: `as68k`, `ld68k`, and `hexdump`. 
This error always happens when `CONFIG_WANXL_BUILD_FIRMWARE == 'y'`.
According to the qualitative analysis of the error message, it says that with the option `CONFIG_PREVENT_FIRMWARE_BUILD='y'` it compiles, however I wasn't able to check this hypothesis once we don't have any configuration with both options set to yes. If we look at the decision tree, it shows that the activation of option `CONFIG_VIDEO_RCAR_VIN` avoids most of the compilation failures (see this issue #137). However, as two errors are still there, It doesn't seem to be a complete explanation of non-compilation failures. Also, this failure cannot be explained by the option `CONFIG_PREVENT_FIRMWARE_BUILD`, since this option is ALWAYS set to `'false'` when `CONFIG_WANXL_BUILD_FIRMWARE` is `yes`. Then, these failures are avoided by enabling (a set of) other(s) configuration option(s), but which one? isn't this option presented in the decision tree? (439 errors)

1. `'aicasm_symbol.c:48:19: fatal error: aicdb.h: No such file or directory', '/bin/sh: 1: as68k: not found'`: It's a combination of the two previous errors. Thus, it's due to `CONFIG_WANXL_BUILD_FIRMWARE == 'y' AND (CONFIG_AIC7XXX_BUILD_FIRMWARE == 'y' or CONFIG_AIC79XX_BUILD_FIRMWARE == 'y')`. (107 errors)

1. `""./include/linux/string.h:305:4: error: call to '__read_overflow2' declared with attribute error: detected read beyond size of object passed as 2nd parameter""`: (thanks to the qualitative analysis of the bug and the decision tree) when making the query `rawtuxdata.query("cluster_error == 'L2'").groupby(['FORTIFY_SOURCE', 'IPV6', 'UBSAN_NULL', 'UBSAN_ALIGNMENT', 'UBSAN_SANITIZE_ALL', 'INFINIBAND_ADDR_TRANS']).count()[['cid']] `, we can find that the error always occur when `FORTIFY_SOURCE == 'y' & IPV6 == 'n' & UBSAN_SANITIZE_ALL == 'y' & INFINIBAND_ADDR_TRANS == 'y'`, consequently the options `'UBSAN_NULL'` and`'UBSAN_ALIGNMENT'` are misclassified by the decision tree. According to the qualitative analysis of the error message, it's due a kernel Linux bug. (82 errors)

To summarize, only part of the errors of (1) and (3) are in fact due option `CONFIG_AIC7XXX_BUILD_FIRMWARE`, the other errors are due other options: `'CONFIG_AIC79XX_BUILD_FIRMWARE'`, `'CONFIG_WANXL_BUILD_FIRMWARE'`, `'FORTIFY_SOURCE'`, `'IPV6'`, `'UBSAN_SANITIZE_ALL'`, `'INFINIBAND_ADDR_TRANS'`.

**Comment by FAMILIAR-project** [comment]
Thanks, see also issues #142 #147 (for overflow2 error) and #137 (for WANXL) it seems consistent with your conclusions.

**Comment by FAMILIAR-project** [comment]
Is `GENERIC_ALLOCATOR + DRM_VBOXVIDEO` bug dominated by `AIC7XXX_BUILD_FIRMWARE` or `AIC79XX_BUILD_FIRMWARE`? 

There are interesting cases we can look at: 
`rawtuxdata.query("cid >= 30000 & GENERIC_ALLOCATOR == 0 & DRM_VBOXVIDEO == 2 & (AIC7XXX_BUILD_FIRMWARE == 1 | AIC79XX_BUILD_FIRMWARE == 1)")[['cid', compilation_status_column_name, 'AIC7XXX_BUILD_FIRMWARE']]`

|    cid |   compile_success |   AIC7XXX_BUILD_FIRMWARE |   AIC79XX_BUILD_FIRMWARE |   GENERIC_ALLOCATOR |   DRM_VBOXVIDEO |
| ------:|------------------:|-------------------------:|-------------------------:|--------------------:|----------------:|
|  30901 |                 0 |                        1 |                        0 |                   0 |               2 |
|  37341 |                 0 |                        1 |                        0 |                   0 |               2 |
|  40757 |                 0 |                        1 |                        1 |                   0 |               2 |
|  48193 |                 0 |                        1 |                        0 |                   0 |               2 |
|  50359 |                 0 |                        1 |                        0 |                   0 |               2 |
|  55986 |                 0 |                        1 |                        0 |                   0 |               2 |
|  59253 |                 0 |                        1 |                        1 |                   0 |               2 |
|  61758 |                 0 |                        0 |                        1 |                   0 |               2 |
|  67576 |                 0 |                        0 |                        1 |                   0 |               2 |
|  79099 |                 0 |                        1 |                        0 |                   0 |               2 |
|  84168 |                 0 |                        0 |                        1 |                   0 |               2 |

and all have 'AICDB' or 'AIC7XXX' error message.
